front/components/poke/subscriptions/EnterpriseUpgradeDialog.tsx

```diff
@@ -0,0 +1,250 @@
+                    </PokeSelectItem>
+                  ))}
+                </div>
+              </PokeSelectContent>
+            </PokeSelect>
+          </PokeFormControl>
+          <PokeFormMessage />
+        </PokeFormItem>
+      )}
+    />
+  );
+}
+
+function InputField({
+  control,
+  name,
+  title,
+  type,
+  placeholder,
+}: {
+  control: Control<EnterpriseUpgradeFormType>;
+  name: keyof EnterpriseUpgradeFormType;
+  title?: string;
+  type?: "text" | "number";
+  placeholder?: string;
+}) {
+  return (
+    <PokeFormField
+      control={control}
+      name={name}
+      render={({ field }) => (
+        <PokeFormItem>
+          <PokeFormLabel className="capitalize">{title ?? name}</PokeFormLabel>
+          <PokeFormControl>
+            <PokeInput
+              placeholder={placeholder ?? name}
+              type={type}
+              {...field}
+              value={field.value}
+            />
+          </PokeFormControl>
+          <PokeFormMessage />
+        </PokeFormItem>
+      )}
+    />
+  );
+}
+
+export default function EnterpriseUpgradeDialog({
+  disabled,
+  owner,
+}: {
+  disabled: boolean;
+  owner: WorkspaceType;
+}) {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [open, setOpen] = useState(false);
+
+  const { plans } = usePokePlans();
+
+  const form = useForm<EnterpriseUpgradeFormType>({
+    resolver: ioTsResolver(EnterpriseUpgradeFormSchema),
+    defaultValues: {
+      stripeSubscriptionId: "",
+      planCode: "",
+    },
+  });
+
+  const onSubmit = (values: EnterpriseUpgradeFormType) => {
+    const cleanedValues = Object.fromEntries(
+      removeNulls(
+        Object.entries(values).map(([key, value]) => {
+          if (typeof value !== "string") {
+            return [key, value];
+          }
+          const cleanedValue = value.trim();
+          if (!cleanedValue) {
+            return null;
+          }
+          return [key, cleanedValue];
+        })
+      )
+    );
+
+    const submit = async () => {
+      setIsSubmitting(true);
+      setError(null);
+      try {
+        const r = await fetch(
+          `/api/poke/workspaces/${owner.sId}/upgrade_enterprise`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify(cleanedValues),
+          }
+        );
+
+        if (!r.ok) {
+          throw new Error(
+            `Something went wrong: ${r.status} ${await r.text()}`
+          );
+        }
+
+        form.reset();
+        setOpen(false);
+      } catch (e) {
+        setIsSubmitting(false);
+        if (e instanceof Error) {
+          setError(e.message);
+        }
+      }
+    };
+    void submit();
+  };
+
+  return (
+    <PokeDialog open={open} onOpenChange={setOpen}>
+      <PokeDialogTrigger asChild>
+        <PokeButton variant="outline" disabled={disabled}>
```

Comment:
NIT: Could we have the button not full width? 

Super NIT: There's now two different style with Poké and Sparkle buttons on the same modal. Could we only use Poké Buttons since we introduced them?


Super Giga NIT that definitely could be in another PR: Should we migrate all Poké Sparkleq modals to the new PokeDialog?

Reply:
Let me use PokeButton everywhere.

Reply:
Done